plugins/package-managers/maven/src/main/kotlin/utils/MavenSupport.kt

```diff
@@ -310,6 +313,57 @@ class MavenSupport(private val workspaceReader: WorkspaceReader) {
                 runCatching { Hash(it, algorithm) }.getOrNull()
             } ?: Hash.NONE
 
+        /**
+         * Convert this [RemoteRepository] to a repository in the format used by the Maven Repository System.
+         * Make sure that all relevant properties are set, especially the proxy and authentication.
+         */
+        internal fun RemoteRepository.toArtifactRepository(
+            repositorySystemSession: RepositorySystemSession,
+            repositorySystem: MavenRepositorySystem,
+            id: String
+        ) = repositorySystem.createRepository(url, id, true, null, true, null, null).apply {
+            this@toArtifactRepository.proxy?.let { repoProxy ->
+                proxy = Proxy().apply {
+                    host = repoProxy.host
+                    port = repoProxy.port
+                    protocol = repoProxy.type
+                    toMavenAuthentication(
+                        AuthenticationContext.forProxy(
+                            repositorySystemSession,
+                            this@toArtifactRepository
+                        )
+                    )?.let { authentication ->
```

Comment:
As the return value does not seem to be used, would `also` be a better fit than `let`?

Reply:
If you want me, I can of course change this and the other occurrences. However, I think it is a rather typical pattern in this and other related codebases to use `let` for applying side-effects on a nullable value.

Reply:
> However, I think it is a rather typical pattern in this and other related codebases to use `let` for applying side-effects on a nullable value.

Actually, "applying side-effects" [is exactly the distinguishing factor of `let` vs. `also`](https://www.digitalocean.com/community/tutorials/kotlin-let-run-also-apply-with#kotlin-also): You should use `also` if you *also* want to do somehting on the object as a side-effect, but are not after mapping its value like `let` does.

So to stick to those semantics, I'd indeed prefer to use `also` instead of `let` here.

Reply:
Sure